rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -120,8 +120,8 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
       }
     }
     for (size_t i = 0; i < waitable_handles_.size(); ++i) {
-      if (!waitable_handles_[i]->is_ready(wait_set)) {
-        waitable_handles_[i].reset();
+      if (waitable_handles_[i]->is_ready(wait_set)) {
+        waitable_triggered_handles_.emplace_back(waitable_handles_[i]);
```

Comment:
```suggestion
        waitable_triggered_handles_.emplace_back(std::move(waitable_handles_[i]));
```

and add `#include <utility>` before.